modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/DynamicChannelLoaderImpl.java

```diff
@@ -0,0 +1,114 @@
+package org.motechproject.tasks.service.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.motechproject.tasks.domain.DynamicChannelProvider;
+import org.motechproject.tasks.domain.TaskTriggerInformation;
+import org.motechproject.tasks.domain.TriggerEvent;
+import org.motechproject.tasks.service.DynamicChannelLoader;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class DynamicChannelLoaderImpl implements DynamicChannelLoader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicChannelLoaderImpl.class);
+
+    @Autowired
+    private BundleContext bundleContext;
+
+    public List<TriggerEvent> getDynamicTriggers(String moduleName, int page, int pageSize) {
+        LOGGER.debug("Retrieving dynamic triggers for channel {}", moduleName);
+
+        DynamicChannelProvider provider = getChannelProvider(moduleName);
+
+        if (provider != null) {
+            return provider.getTriggers(page, pageSize);
+        }
+
+        return new ArrayList<>();
+    }
+
+    public boolean providesDynamicTriggers(String moduleName) {
+        LOGGER.debug("Retrieving dynamic triggers for channel {}", moduleName);
+
+        DynamicChannelProvider provider = getChannelProvider(moduleName);
+
+        return provider != null;
+    }
+
+    public Long countByChannelModuleName(String moduleName) {
+        DynamicChannelProvider provider = getChannelProvider(moduleName);
+
+        if (provider != null) {
```

Comment:
We should probably log something if there's no provider. This might save someone some debugging time